packages/odyssey-react-mui/src/themes/odyssey/components.ts

```diff
@@ -0,0 +1,321 @@
+            borderColor: "#ebebed",
+            backgroundColor: "#ebebed",
+            color: "#8c8c96",
+          },
+        },
+      },
+      {
+        props: { variant: "danger" },
+        style: {
+          backgroundColor: "#da372c",
+          color: "#ffffff",
+          "&:hover": {
+            borderColor: "transparent",
+            backgroundColor: "#640019",
+          },
+
+          "&:focus-visible": {
+            outlineColor: "#f88c90",
+            backgroundColor: "#640019",
+          },
+
+          "&:active": {
+            borderColor: "transparent",
+            backgroundColor: "#da372c",
+          },
+
+          "&:disabled": {
+            color: "#ffffff",
+            borderColor: "#f88c90",
+            backgroundColor: "#f88c90",
+          },
+        },
+      },
+      {
+        props: { variant: "floating" },
+        style: {
+          backgroundColor: "#ffffff",
+          color: "#1d1d21",
+          borderColor: "transparent",
+
+          "&:hover, &:focus-visible": {
+            backgroundColor: "rgba(29, 29, 33, 0.1)",
+            borderColor: "transparent",
+          },
+          "&:active": {
+            backgroundColor: "rgba(29, 29, 33, 0.2)",
+            borderColor: "transparent",
+          },
+          "&:disabled": {
+            backgroundColor: "rgba(235, 235, 237, 0.6)",
+            color: "#6e6e78",
+            borderColor: "transparent",
+          },
+        },
+      },
+      {
+        props: { size: "s" },
+        style: {
+          paddingBlock: "calc(0.57142857rem - 1px)",
+          paddingInline: "calc(0.85714286rem - - 1px)",
+          fontSize: "1rem",
+        },
+      },
+      {
+        props: { size: "l" },
+        style: {
+          paddingBlock: "calc(1.14285714rem - 1px)",
+          paddingInline: "calc(1.14285714rem - 1px)",
+        },
+      },
+      {
+        props: { fullWidth: true },
+        style: {
+          display: "block",
+          width: "100%",
+          marginBlock: "0",
+          marginInline: "0",
+
+          "&:not(:last-child)": {
+            marginBlockEnd: "1.14285714rem",
+          },
+        },
+      },
+    ],
+    styleOverrides: {
+      root: ({ ownerState }) => ({
+        ...ownerState,
+        fontWeight: 600,
+        paddingBlock: "0.85714286rem",
+        paddingInline: "0.85714286rem",
+        display: "inline-block",
+        position: "relative",
+        marginBlock: "0",
+        marginInline: "0",
+        transitionProperty:
+          "color, background-color, border-color, outline-offset, outline-color",
+        transitionDuration: "100ms",
+        transitionTimingFunction: "linear",
+        borderWidth: "1px",
+        borderStyle: "solid",
+        outlineColor: "transparent",
+        outlineOffset: "0",
+        fontSize: "1rem",
+        lineHeight: "1.14285714",
+        whiteSpace: "nowrap",
+
+        "& + &": {
+          marginInlineStart: "0.57142857rem",
+        },
```

Comment:
I don't like this built-in margin styling. It can cause a lot of edge case bugs. I think we have this right now for `Button`, but I would argue it's not the responsibility of a button to do that, it's the responsibility of a `ButtonGroup` component.

Reply:
Agreed, we plan on removing this soon.